Homework2/CoherentTrainingSystem/Training.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace CoherentTrainingSystem
+{
+    public class Training : TrainingBase
+    {
+        public List<TrainingBase> LecturesAndPractice = new List<TrainingBase>();
+
+        public void Add(TrainingBase training)
+        {
+            LecturesAndPractice.Add(training);
+        }
+
+        public void Add(List<TrainingBase> trainings)
```

Comment:
make incoming type as IEnumerable<TrainingBase>. 
This is a best practice, to caller of the method can pass array, list or any other inheritor of IEnumerable